src/utils/work/fetchWork.js

```diff
@@ -68,20 +68,22 @@ const fetchGeneExpressionWork = async (
   // Then, we may be able to find this in S3.
   let response = await seekFromS3(ETag, experimentId);
 
+  // If there is no response in S3, dispatch workRequest via the worker
   if (!response) {
-    response = await seekFromAPI(
+    await dispatchWorkRequest(
       experimentId,
       missingGenesBody,
       timeout,
       ETag,
-      null,
       {
         ETagPipelineRun: qcPipelineStartDate,
         ...extras,
       },
     );
   }
 
+  response = await seekFromS3(ETag, experimentId);
+
   if (!response[missingGenes[0]]?.error) {
     // Preprocessing data before entering cache
     const processedData = calculateZScore(response);
```

Comment:
why do you calculate zscore here?

Reply:
After discussing with @ogibson-biomage and @cosa65, calculation of the z-score is still left in the UI because we still have to work on passing. Currently, gene expression data sent from the worker to the UI containing the truncated data and the raw data is quite big (300 MB). If we add z-score along with the data, it will increase the work request size by about 50%, making downloading the work request slower for our users with slow connection.

The gene expression data is actually sparse (e.g. contain a lot of `null` values). We use a lot of space to still include these `null` values to make it easy to access the gene expression data. However, this is an issue 

I've actually created a ticket for this (https://biomage.atlassian.net/browse/BIOMAGE-1710) before the discussion with Martin and Oliver. Oliver argues that this ticket should not be done before we address the gene expression work request. He is going to add the reason to **not** do this right now in the ticket, but as a downstream ticket after addressing the gene expression work request issues.